### PR TITLE
docs: rework components/data fetching pages

### PR DESCRIPTION
**Pull request by danielroe**
**TODO**:
- [x] Adding "Features > Components Discovery" section
- [x] Renaming "Features > Built-in Components"
- [x] Update "Directory Structure > Components" with components v2 docs
- [x] Update "Configuration Glossary > components" with detailed version of options
- [x] ~~Apply and~~ mention conventions based on vue-style-guide (for base components, naming, etc)

### CODE REVIEW

**Comment by Atinux** [comment]
Nice work, we now need to add the configuration glossary into https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-components

**Comment by pi0** [comment]
Ref: #1244

**Comment by inouetakuya** [comment]
@danielroe 

I found CI failure (textlint for Japanese) and have fixed on the following PR.

- [fix: textlint for Japanese by inouetakuya · Pull Request #1439 · nuxt/nuxtjs.org](https://github.com/nuxt/nuxtjs.org/pull/1439)

File: content/en/guides/configuration-glossary/configuration-components.md [hunk 1]
```diff
@@ -8,13 +8,253 @@ position: 5
 
 > Nuxt.js 2.13+ can scan and auto import your components.
 
+<base-alert>It is possible to use this feature with Nuxt 2.10 - 2.12. Just manually install and add `@nuxt/components` to `buildModules` inside `nuxt.config`.</base-alert>
```
Comment (pi0) [inline content/en/guides/configuration-glossary/configuration-components.md:11]:
```suggestion
<base-alert>It is possible to use this feature with Nuxt 2.10 - 2.12 if you cannot upgrade. Just manually install and add `@nuxt/components` to `buildModules` inside `nuxt.config`.</base-alert>
```

**Review by pi0 — DISMISSED** [review]

File: content/en/guides/configuration-glossary/configuration-components.md (hunk 1 above)
Comment (pi0) [reply content/en/guides/configuration-glossary/configuration-components.md:11]: (or we can just not mention this to be consistent with features section that is saying 2.13+ needed)

**Review by Atinux — DISMISSED** [review]

**Comment by Atinux** [comment]
Shall we merge @danielroe ?

**Comment by danielroe** [comment]
I will likely need to rebase this against current main branch.

**Pull request closed without merging**